src/transformers/models/deit/modeling_deit.py

```diff
@@ -399,15 +399,20 @@ class DeiTPreTrainedModel(PreTrainedModel):
     main_input_name = "pixel_values"
     supports_gradient_checkpointing = True
 
-    def _init_weights(self, module: Union[nn.Linear, nn.Conv2d, nn.LayerNorm]) -> None:
+    def _init_weights(self, module) -> None:
         """Initialize the weights"""
         if isinstance(module, (nn.Linear, nn.Conv2d)):
-            module.weight.data = nn.init.trunc_normal_(module.weight.data, mean=0.0, std=self.config.initializer_range)
+            torch.nn.init.trunc_normal_(module.weight.data, std=self.config.initializer_range)
```

Comment:
FYI @alaradirik, if a PyTorch method has _ at the end, it means the operation is done in-place